backend/isisInternal.py

```diff
@@ -189,8 +190,22 @@ def getAllFieldJournals(instrument, field, search):
         tree = ET.parse(response)
         root = tree.getroot()
         fields = []
-        path = "//*[contains(data:"+field+",'"+search+"')]"
+        if field == "run_number":
+            values = search.split("-")
+            path = "//*[data:run_number>"+values[0] + \
+                " and data:run_number<"+values[1]+"]"
+        if field == "start_date":
```

Comment:
Shouldn't this be an `else if`? Otherwise the run_number clause can run in addition to the else of the second check.